### PR TITLE
Internal data branch

### PR DESCRIPTION
**Pull request by rupaliT123**

### CODE REVIEW

**Comment by rupaliT123** [comment]
dev:
lint:0
coverage:85%
internal_data_branch:
lint:0
coverage:85%

File: src/test/java/com/techsophy/tsf/account/controller/InternalControllerImplTest.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    @Mock
+    UserFormDataDefinitionRepository userFormDataRepository;
+    @Mock
+    UserServiceImpl userServiceImpl;
+    @Mock
+    ObjectMapper objectMapper;
+    @Mock
+    GlobalMessageSource globalMessageSource;
+    @Mock
+    IdGeneratorImpl idGenerator;
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/account/controller/InternalControllerImplTest.java:52]: remove all the unnecessary mocks

File: src/main/java/com/techsophy/tsf/account/controller/impl/InternalControllerImpl.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    @Value(ENCRYPTION_KEY_FILE)
+    String keyLocation;
+    @Override
+    public ApiResponse<UserFormDataSchema> saveUser(UserFormDataSchema internalUserFormDataSchema, HttpHeaders headers) throws JsonProcessingException
+    {
+        try
+        {
+            if(headers.containsKey(X_SIGNATURE)) {
+                String headerSign = headers.getFirst(X_SIGNATURE);
+                Rsa4096 rsa4096 = new Rsa4096(keyLocation);
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/controller/impl/InternalControllerImpl.java:36]: This class need not be instantiated per request and should be moved as a member variable

File: src/main/java/com/techsophy/tsf/account/utils/Rsa4096.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    public UserFormDataSchema transform(String headerSign,UserFormDataSchema internalUserFormDataSchema)
+    {
+
+        try
+        {
+            String signatureValue = (String)internalUserFormDataSchema.getUserData().get(USERNAME_VERIFICATION);
+            Boolean isVerified = verifySignature(headerSign,signatureValue);
+            if(isVerified != null && !isVerified)
+            {
+                throw new InvalidDataException(UN_AUTHORIZED_EXCEPTION,globalMessageSource.get(UN_AUTHORIZED_EXCEPTION));
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/utils/Rsa4096.java:88]: pleaes add test cases for each of these exception that are being thrown.

File: src/test/java/com/techsophy/tsf/account/controller/InternalControllerImplTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.techsophy.tsf.account.constants.AccountConstants.*;
+import static com.techsophy.tsf.account.constants.PropertyConstant.*;
+
+@ExtendWith({MockitoExtension.class, SpringExtension.class})
+class InternalControllerImplTest {
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/account/controller/InternalControllerImplTest.java:42]: Please add test cases when
- signature is not present
- signature is not verified
- signature is verified
- if user service throws an exception
- etc

File: src/main/java/com/techsophy/tsf/account/controller/impl/InternalControllerImpl.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import static com.techsophy.tsf.account.constants.AccountConstants.*;
+import static com.techsophy.tsf.account.constants.PropertyConstant.X_SIGNATURE;
+
+@RestController
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
+public class InternalControllerImpl implements InternalController {
+    private final UserFormDataService userFormDataService;
+    private final GlobalMessageSource globalMessageSource;
+    @Value(ENCRYPTION_KEY_FILE)
+    String keyLocation;
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/controller/impl/InternalControllerImpl.java:28]: this is keycloak, public key location name it appropriately

File: src/test/java/com/techsophy/tsf/account/controller/InternalControllerImplTest.java (hunk 4 above)
Comment (VaibhavTechsophy) [reply src/test/java/com/techsophy/tsf/account/controller/InternalControllerImplTest.java:42]: If key is not found
key is invalid

File: src/main/java/com/techsophy/tsf/account/utils/Rsa4096.java [hunk 6]
```diff
@@ -78,22 +77,17 @@ public Boolean verifySignature(String signature, String value) throws SignatureE
 
     public UserFormDataSchema transform(String headerSign,UserFormDataSchema internalUserFormDataSchema)
     {
-
         try
         {
-            String signatureValue = (String)internalUserFormDataSchema.getUserData().get(USERNAME_VERIFICATION);
+            String signatureValue = (String)internalUserFormDataSchema.getUserData().get(USERNAME_INTERNAL);
```
[the remark this answers is not in the document]
Comment (VaibhavTechsophy) [reply src/main/java/com/techsophy/tsf/account/utils/Rsa4096.java:82]: Clarified with Sanjay that only username value is used to sign

**Review by VaibhavTechsophy — APPROVED** [review]

**Comment by VaibhavTechsophy** [comment]
Please coordinate with Nandini/Akhil to deploy latest changes

**Pull request merged**